### ISSUE TITLE
Freeze & crash Windows Terminal Preview 1.21.1272.0 with these few simple steps, you'll never guess what the 3rd one is!

### ISSUE THREAD

**Issue by PhMajerus** [microsoft/terminal#17202]
### Windows Terminal version

1.21.1272.0

### Windows build number

10.0.22631.3527 x64

### Other Software

Just Terminal Preview, crashes with any shell (testes cmd.exe, bash in Ubuntu in WSL2).

### Steps to reproduce

1. Open Windows Terminal Preview with your default shell
2. Click [v] menu, and select Settings to display the settings page
3. Tear-off the Settings panel into its own window, making both settings and your shell visible
4. Select your shell profile, click [Appearance >], toggle Builtin Glyphs off, click [Save], toggle Builtin Glyphs back on, click [Save]
5. Close the Settings tab, which closes its individual window.
6. Wait ~30 seconds (could be longer apparently if focus is changed)
7. Give the focus to some other app, then click on your shell panel to give it the focus again
8. ?
9. Profit!

### Expected Behavior

Windows Terminal should keep running

### Actual Behavior

Windows Terminal freezes, its client area gets ghosty transparent, it receives the dreaded `(Not Responding)` suffix, and Dr. Watson comes along to put it out if its misery.

**Comment by DHowett** [comment]
Dr. Watson as the grimmest reaper. I love it.

Would you be able to file feedback and reproduce the crash in the Apps > Terminal node and share the link with us?

I can't get this to happen locally . . . and I want to see if it's the same bug we just fixed with #17199.

**Comment by PhMajerus** [comment]
Feedback Hub: https://aka.ms/AAqbvpz
I'm not sure it captured the proper dump though, only `Windows Terminal` shows in the apps list, no separate entry for `Windows Terminal Preview`.
Let me know.

**Comment by PhMajerus** [comment]
I can get it to freeze every single time on a Surface Book 2 15", but cannot reproduce on an Intel NUC (NUC13ANKI50WC).

Could it be something GPU-related? The Surface Book 2 has always been a bit unreliable, even Edge sometimes restarts its rendering engine.

**Comment by DHowett** [comment]
I'm seeing a thread tearing down a `TSF::Implementation` for a handwriting input method, a couple shell worker threads also stuck tearing down, and a window trying to update its acrylic background policy. That's weird.

Watson cab `97512468-57cc-48ca-97d5-9721b4e3ec53`

**Comment by PhMajerus** [comment]
It's really weird, I could narrow it down to showing the appearance page for any shell, not necessarily the one being used in the other window.
But the settings has to be the tab tore off, if it's another shell window and then that window is used to show the settings and do everything similarly, then it won't crash.

So it has to be the settings page that is tore off to create a new window, and it has to show the appearance with the preview. You can navigate to another setting page again, as long as the preview has been displayed, it will crash.
If the settings is shown in the primary window or in a window created by tearing off another shell tab, it doesn't crash.

Is there some Atlas resources that are shared with the preview and may be freed when Settings is closed? Why would it only happen when the settings is tore off?

**Comment by PhMajerus** [comment]
Could this be related?

![image](https://github.com/microsoft/terminal/assets/25664275/9aa0fc85-009a-4cf6-8373-88a1d5a82cb9)

I know Microsoft Surface Book 2 display drivers are somewhat unreliable, and Microsoft's Surface team never fixed the issue, Edge also sometimes has to restart its rendering, blacking out all rendering windows and recovering.

**Comment by PhMajerus** [comment]
Attached another crash dump to the Feedback Hub report.
This one does not even freeze, it crashes directly, and doesn't involve the Settings page, it's just opening an Ubuntu tab, doing `apt update/upgrade/autoremove/autoclean`, then `exit` to close the tab. Terminal shows the original cmd tab again and crashes immediately.

**Comment by lost22git** [comment]
I just updated this version and there is also a problem.

The prompt will disappear inexplicably and the cursor will move to the bottom.

If you don't press a key, everything goes blank.

![2](https://github.com/microsoft/terminal/assets/65008815/c3ff93c4-b4e9-46b1-87c6-1df2c86db326)

**Comment by scc23456** [comment]
try to disable "Shell handwriting" option in Windows Settings...

**Comment by PhMajerus** [comment]
@scc23456 Good catch!
Indeed disabling the `Shell handwriting` feature seems to fix both crashes, and that would explain the shell handwriting input method that DHowett saw in the crash dump.

So there seems to be a problem between Terminal and "write anywhere", and this shows why it is important to try to test software on some devices with 👆 touch and 🖊️ pen, not only on typical workstations.

**Comment by PhMajerus** [comment]
> Does this only work with pens?

Yeah, it's a recent feature the shell team introduced to write directly in many common controls. For example we can ink directly inside a textbox or address bar using the pen.

![image](https://github.com/microsoft/terminal/assets/25664275/4e9daeed-0d13-411c-8324-9c93535d305a)

This is intended to be a faster way to input text without having to display the tablet input panel (TabTip).
You'll need a device with a pen digitizer to test it out.

I suspect you're using or subclassing some common control, the shell attaches some overlay inking panel on top of it, and you destroy it without going through the original code that ensures the attached inking panel is properly disposed of, then it crashes when trying to access its parent HWND or something. This is just a wild guess though, I haven't checked how they implemented the Ink Anywere feature yet.

**Comment by PhMajerus** [comment]
@lhecker Are you giving up on this one? I didn't see any similar crash with that feature in other apps, there is probably something Terminal does that makes it crash when that feature extends its controls.

**Comment by DHowett** [comment]
> Are you giving up on this one? 

Please don't read too much into how we manage our issue assignments. 😅

**Comment by DHowett** [comment]
Good news! We couldn't reproduce this because it's fixed in Windows 11 Build 26100 (24H2 release). It still reproduces in the previous release.

Tracking - MSFT:46832480

**Comment by DHowett** [comment]
This fix should have rolled out by now :)
